### PR TITLE
SNOW-1820302 Integrate with ingest sdk 3.0.1

### PR DESCRIPTION
**Pull request by sfc-gh-mbobowski**
# Overview

SNOW-1820302

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

**Review by sfc-gh-rsawicki — APPROVED** [review]

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergTableSchemaResolver.java [hunk 1]
```diff
@@ -61,26 +61,13 @@ private IcebergColumnSchema mapIcebergSchemaFromChannel(
       Map.Entry<String, ColumnProperties> schemaFromChannelEntry) {
 
     ColumnProperties columnProperty = schemaFromChannelEntry.getValue();
-    String plainIcebergSchema = getIcebergSchema(columnProperty);
+    String plainIcebergSchema = columnProperty.getIcebergSchema();
```
Comment (sfc-gh-bzabek) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergTableSchemaResolver.java:64]: +1

**Review by sfc-gh-wtrefon — APPROVED** [review]

**Review by sfc-gh-dseweryn — APPROVED** [review]

File: src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java [hunk 2]
```diff
@@ -261,7 +256,7 @@ public void testComplexRecordEvolution() throws Exception {
           new DescribeTableRow("APPROVAL", "BOOLEAN"),
           new DescribeTableRow("ARRAY1", "ARRAY(NUMBER(19,0))"),
           new DescribeTableRow("ARRAY2", "ARRAY(VARCHAR(16777216))"),
-          new DescribeTableRow("ARRAY3", "ARRAY(BOOLEAN)"),
+          new DescribeTableRow("ARRAY3", "ARRAY(VARCHAR(16777216))"),
```
Comment (sfc-gh-bzabek) [inline src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java:259]: Sry I missed that somehow.

Comment (sfc-gh-mbobowski) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java:259]: No problem, it happens without the CI.

Comment (sfc-gh-wtrefon) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java:259]: funny thing, I didn't notice it in my PR as this is the expected result. Now this breaks my PR, have to rollback this line :D

Comment (sfc-gh-wtrefon) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java:259]: now i feel sad that i find such things "funny"

**Pull request merged**